packages/backend/src/vm-manager.ts

```diff
@@ -0,0 +1,200 @@
+/**********************************************************************
+ * Copyright (C) 2024 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import path from 'node:path';
+import * as extensionApi from '@podman-desktop/api';
+import { isArm, isMac } from './machine-utils';
+import fs from 'node:fs';
+import type { BootcBuildInfo } from '/@shared/src/models/bootc';
+
+// Singular pid file location (can only run 1 VM at a time)
+// eslint-disable-next-line sonarjs/publicly-writable-directories
+const pidFile = '/tmp/qemu-podman-desktop.pid';
+
+// MacOS related
+const macQemuArm64Binary = '/opt/homebrew/bin/qemu-system-aarch64';
+const macQemuArm64Edk2 = '/opt/homebrew/share/qemu/edk2-aarch64-code.fd';
+const macQemuX86Binary = '/opt/homebrew/bin/qemu-system-x86_64';
+
+// Default values for VM's
+const hostForwarding = 'hostfwd=tcp::2222-:22';
+const memorySize = '4G';
+const websocketPort = '45252';
+const rawImageLocation = 'image/disk.raw';
+
+export default class VMManager {
+  private build: BootcBuildInfo;
+
+  // Only values needed is the location of the VM file as well as the architecture of the image that
+  // will be used.
+  constructor(build?: BootcBuildInfo) {
+    this.build = build!;
```

Comment:
please don't do that

it can't be optional and then you're saying there is a value with !

it will make unexpected runtime failures